pkg/state/appender.go

```diff
@@ -628,6 +631,7 @@ func (a *txAppender) appendBlock(params *appendBlockParams) error {
 			consensusImprovementsActivated: consensusImprovementsActivated,
 			invokeExpressionActivated:      invokeExpressionActivated,
 			validatingUtx:                  false,
+			txPosInBlock:                   uint64(i + 1),
```

Comment:
This assumption is incorrect. The `txPosInBlock` field indicates the issuance position of an asset within a block. It can be incremented by either the `IssueTransaction` or the `IssueAction` of an `InvokeTransaction`. Perhaps the field's name does not accurately reflect its meaning. However, it should be noted that other transactions and actions do not modify this value.